### PR TITLE
[#75]  Show loading states - Details

### PR DESCRIPTION
**Pull request by Tumne**
Completed a subtle loading state spinner for sections within player details. Chose to go with the scaleLoader in https://www.davidhu.io/react-spinners/ .  Make sure to `npm install` prior to `npm start` as I've added in a bunch of third party libraries.

UI Suggestions welcome.

Related Issue: #75 

---

Screen Shot 2020-05-20 at 12 40 08 AM

---

### What's Changed:

- quickStats, totalStats, recentGames spinners
- loading state handled in `playerDetailsReducer.js` (formerly `getSelectedPlayerReducer.js`)
- Created shared spinner component (used [react-spinners](https://github.com/davidhu2000/react-spinners) and [emotion](https://emotion.sh/docs/introduction))
- Included proptype checking in `_shared/spinner.js`: [prop-types tutorial](https://reactjs.org/docs/typechecking-with-proptypes.html)
- Added a `jsconfig.js` file that will now allow absolute imports 🚀 : [absolute imports tutorial](https://dev.to/mr_frontend/absolute-imports-in-create-react-app-3ge8)
- Removed `getGamesReducer.js` and refactored `playerDetailsReducer.js`

### CODE REVIEW

File: src/actions/actions.js [hunk 1]
```diff
@@ -11,9 +11,10 @@ export const getTeams = () => async dispatch => {
 };
 
 export const getSelectedTeam = team => async dispatch => {
+
     dispatch({ type: 'GET_SELECTED_TEAM', payload: team });
     dispatch({ type: 'GET_TEAM_COLOR', payload: team.primaryColor });
-    dispatch({ type: 'PRELOAD_PLAYER', payload: null });
+    dispatch({ type: 'PRELOAD_PLAYER_DETAILS', payload: null });
```
Comment (kammanz) [inline src/actions/actions.js:17]: You don't need to dispatch this here. We'll dispatch it in the getSelectedPlayer action.

File: src/components/players/details/quickStats/index.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
 
-const QuickStats = ({ quickStats, teamColor }) => {
-    return (
-        <section className={styles.quickStats}>
-            {quickStats.map(({title, value}) => {
-                return (
-                    <div key={title} className={styles.container}>
-                        <div style={{ backgroundColor: `${teamColor}`}} className={styles.header}>{title}</div>
+const QuickStats = ({ quickStats, teamColor, isLoading }) => {
+  console.log(isLoading);
```
Comment (kammanz) [inline src/components/players/details/quickStats/index.js]: Remove console.log

File: src/components/players/details/recentGamesStats/index.module.scss [hunk 3]
```diff
@@ -61,3 +64,7 @@ section {
 .line {
     border-bottom: 1px solid $grey1b;
 }
+
+.spinner {
```
Comment (kammanz) [inline src/components/players/details/recentGamesStats/index.module.scss]: Unused, please delete.

File: src/components/players/details/totalStats/index.js [hunk 4]
```diff
@@ -1,17 +1,21 @@
 import React from 'react';
 
-import Table from '../../../_shared/table';
-import Title from '../../../_shared/title';
+import Spinner from '../../../_shared/spinner';
```
Comment (kammanz) [inline src/components/players/details/totalStats/index.js]: Use absolute import.

File: src/components/_shared/spinner.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import ScaleLoader from 'react-spinners/ScaleLoader';
+import { css } from '@emotion/core';
+import PropTypes from 'prop-types';
+
+import { COLORS } from 'enums';
+
+const Spinner = ({
+  isLoading = false,
+  containerHeight = 48,
```
Comment (kammanz) [inline src/components/_shared/spinner.js]: I know there aren't too many dynamic heights for the spinner but as best practice let's use propTypes to set a default container height. See below.

File: src/components/_shared/spinner.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    color={COLORS.GREY}
+    loading={isLoading}
+    {...props}
+  />
+);
+
+Spinner.propTypes = {
+  isLoading: PropTypes.bool,
+  containerHeight: PropTypes.number,
+};
```
Comment (kammanz) [inline src/components/_shared/spinner.js:28]: We'll add a default container height to the spinner...

`Spinner.defaultProps = {
  containerHeight: 48
};`

**Comment by kammanz** [comment]
Nice work, the loader is subtle and doesn't overemphasize to the user that they're waiting. My only issue from a UI standpoint is how the player details card shrinks in width when a player is selected and the scrollbar appears, then gets wider again on placeholder reset. Maybe on page load we could have a dummy scrollbar track in a faded color?

File: src/components/_shared/spinner.js (hunk 6 above)
Comment (Tumne) [reply src/components/_shared/spinner.js:28]: Nice catch

File: src/actions/actions.js [hunk 7]
```diff
@@ -11,9 +11,10 @@ export const getTeams = () => async dispatch => {
 };
 
 export const getSelectedTeam = team => async dispatch => {
+
```
Comment (Tumne) [inline src/actions/actions.js]: Remove space

File: src/components/_shared/table.module.scss [hunk 8]
```diff
@@ -1,6 +1,7 @@
 @import 'src/scss/variables.scss';
 
 .table {
+    position: relative;
```
Comment (Tumne) [inline src/components/_shared/table.module.scss]: unnecessary `position: relative;`

File: src/components/players/details/totalStats/index.js [hunk 9]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 import Spinner from 'components/_shared/spinner';
 import Table from 'components/_shared/table';
-import Title from 'components/_shared/title';
+import Title from 'components/_shared/title'
```
Comment (Tumne) [inline src/components/players/details/totalStats/index.js]: Missing semi-colon

**Review by Tumne — COMMENTED** [review]
Great review, here are some other "mistakes" that were also in the code. Every once in a while I'll jump on an issue just to get it started (mostly design tweaks) and have you review it instead.

File: src/components/players/details/totalStats/index.js (hunk 9 above)
Comment (kammanz) [reply src/components/players/details/totalStats/index.js]: noted, will make the changes when i merge the code over.

**Comment by Tumne** [comment]
> Nice work, the loader is subtle and doesn't overemphasize to the user that they're waiting. My only issue from a UI standpoint is how the player details card shrinks in width when a player is selected and the scrollbar appears, then gets wider again on placeholder reset. Maybe on page load we could have a dummy scrollbar track in a faded color?

I don't see it on my comp. It might be because your version of windows has the scrollbar in chrome. I would create a separate issue for this.

**Comment by Tumne** [comment]
Ready for review

**Comment by kammanz** [comment]
Looks good, merging into master.

**Pull request merged**